pkg/kv/kvserver/replica_range_lease.go

```diff
@@ -1096,10 +1102,13 @@ func (r *Replica) redirectOnOrAcquireLeaseForRequest(
 			switch status.State {
 			case kvserverpb.LeaseState_ERROR:
 				// Lease state couldn't be determined.
-				log.VEventf(ctx, 2, "lease state couldn't be determined")
+				msg := status.ErrInfo
+				if msg == "" {
+					msg = "lease state could not be determined"
+				}
+				log.VEventf(ctx, 2, "%s", msg)
 				return nil, false, roachpb.NewError(
-					newNotLeaseHolderError(roachpb.Lease{}, r.store.StoreID(), r.mu.state.Desc,
-						"lease state couldn't be determined"))
+					newNotLeaseHolderError(roachpb.Lease{}, r.store.StoreID(), r.mu.state.Desc, msg))
```

Comment:
nit: probably no need to improve this since @knz thought what you have was fine, but if instead of a `string` you attached an `EncodedError`, wouldn't you retain proper redactability across the RPC and wouldn't it also feel "right" to attach an error to a lease with state `ERROR`?

Reply:
I agree with Tobias - I approved this because I knew that Andrei's main thrust was to remove the redaction markers and the PR achieves that, but moving to a structured error would be a nice improvement on top of that.

Reply:
👍🏽 In the interest of keeping things snappy I would merge as-is and we'll keep this in mind for higher-value use cases